papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/controller/ProjectsController.java

```diff
@@ -90,12 +95,75 @@ private Project getValidProject(Long id) {
         List<SprintState> sprintStates = sprintStateMapper.mapFromApi(
                 Optional.ofNullable(sprintStateDtos).orElse(ALL_SPRINT_STATE_DTOS));
 
-        List<SprintDto> sprints = sprintService.findByStatesInProject(sprintStates, project.getId(), currentTime)
+        List<SprintDto> sprints = sprintService.getByStatesInProject(sprintStates, project.getId(), currentTime)
                 .stream()
                 .sorted(Comparator.comparing(sprint -> sprint.getEnrollmentPeriod().getStart()))
                 .map(sprint -> sprintMapper.mapToApi(sprint, currentTime))
                 .collect(Collectors.toList());
 
         return ResponseEntity.ok(sprints);
     }
+
+    @Override
+    public ResponseEntity<AvailabilityDto> getUserAvailability(Long projectId, Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+
+        if (!isSprintInProject(sprint, project)) {
+            throw new ForbiddenAccessException();
+        }
+
+        Availability availability = availabilityService
+                .getBySprintIdAndUserId(sprintId, userContext.getUserIdn())
+                .orElseThrow(ResourceNotFoundException::new);
+
+        return ResponseEntity.ok(availabilityMapper.mapToApi(availability));
+    }
+
+    @Override
+    public ResponseEntity<AvailabilityDto> updateUserAvailability(@Valid AvailabilityDto body, Long projectId,
```

Comment:
`AvailabilityDto body` -> nazwałabym inaczej niż `body` - stare dobre `AvailabilityDto availabilityDto`?

Reply:
Co robi `@Valid`? Mogłabym poszukać, ale mogę też zapytać

Reply:
nie wiem, samo się generuje 
przypuszczam, że to przez nie wywala błąd, jeśli coś required jest nullem